### ISSUE TITLE
libquiche.so undefined references

### ISSUE THREAD

**Issue by bagder** [cloudflare/quiche#1761]
When I run curl's configure to try to use quiche from current git, I get these errors:

~~~
/usr/bin/ld: $path/libquiche.so: undefined reference to `__gxx_personality_v0'
/usr/bin/ld: $path/libquiche.so: undefined reference to `vtable for __cxxabiv1::__si_class_type_info'
/usr/bin/ld: $path/libquiche.so: undefined reference to `operator delete(void*, unsigned long)'
/usr/bin/ld: $path/libquiche.so: undefined reference to `vtable for __cxxabiv1::__class_type_info'

~~~
(`$path` is me hiding my local long path to make it easier to the eye)

Any clues on what I am missing? (These looks like traces of C++, which I suspected might come from BoringSSL, but I am able to build from current BoringSSL without getting this problem.)

**Comment by ghedo** [comment]
Are you using the BoringSSL version vendored by quiche or a different one? Wonder if it's related to https://boringssl-review.googlesource.com/c/boringssl/+/66288 (which would affect newer BoringSSL versions than what quiche currently uses).

**Comment by bagder** [comment]
Ah yes, I have a separate BoringSSL build. I forgot about that. I updated to their latest git just a few days ago (April 12, 2024).

I too strongly suspect it has something to do with that C++ism they introduced.

**Comment by Lekensteyn** [comment]
Interesting issue, I might take a look later. BoringSSL switched all sources to C++ in https://github.com/google/boringssl/commit/5813c2c10c73d800f1b0d890a7d74ff973abbffc but indicated that "the use of the C++ standard library is minimal and no run-time requirement for it is intended."

A workaround is to link to libstdc++ as is done in PR #1973.

Would be nice to update the default boringssl version to at least commit 72a60506d (includes commit 7fb4d3da5) for PQ support based on X25519MLKEM768.

**Comment by pkropachev** [comment]
Yes, linking with `libstdc++` also solves problem building `curl` with the latest version of the BoringSSL. Link: https://github.com/curl/curl/issues/13523#issuecomment-2093096055.

Actually it would be great bump version of SSL for quiche library. Currently used version is pretty old.
